perun-core/src/main/java/cz/metacentrum/perun/core/implApi/DatabaseManagerImplApi.java

```diff
@@ -12,30 +14,60 @@
 	/**
 	 * Return current database version in string (ex. 3.0.1)
 	 * 
-	 * @param perunSession
 	 * @return return current database version
 	 * 
 	 * @throws InternalErrorException
 	 */
-	String getCurrentDatabaseVersion(PerunSession perunSession) throws InternalErrorException;
+	String getCurrentDatabaseVersion() throws InternalErrorException;
 	
 	/**
 	 * Get DB driver information from datasource (name-version)
 	 * 
-	 * @param sess
 	 * @return string information about database driver
 	 * 
 	 * @throws InternalErrorException 
 	 */
-	String getDatabaseDriverInformation(PerunSession sess) throws InternalErrorException;
+	String getDatabaseDriverInformation() throws InternalErrorException;
 	
 	/**
 	 * Get DB information from datasource (name-version)
 	 * 
-	 * @param sess
 	 * @return string information about database
 	 * 
 	 * @throws InternalErrorException 
 	 */
-	String getDatabaseInformation(PerunSession sess) throws InternalErrorException;
+	String getDatabaseInformation() throws InternalErrorException;
+
+	/**
+	 * Returns current code version from DBChangelog.txt (ex. 3.0.1)
+	 *
+	 * @param fileName DB changelog file name, file should be in resources
+	 *
+	 * @return current code version
+	 *
+	 * @throws InternalErrorException
+	 */
+	String getCodeDatabaseVersion(String fileName) throws InternalErrorException;
+
+	/**
+	 * Method updates database to the current code version. Current code version is written in DB changelog file.
+	 *
+	 * @param currentDBVersion current DB version
+	 * @param fileName DB changelog file name, file should be in resources
+	 *
+	 * @throws InternalErrorException
+	 */
+	void updateDatabaseVersion(String currentDBVersion, String fileName) throws InternalErrorException;
+
+	/**
+	 * Parse all database versions from DB changelog file and create list of DBVersion objects.
```

Comment:
Please add information (to all javadocs about this method) that it create list of DBVersion from currentDBVersion to now (without currentDBVersion itself).